python/eos/tasks.py

```diff
@@ -79,6 +81,97 @@ def task_wrapper(*args, **kwargs):
     return _task
 
 
+@task('{posterior}/modes')
+def find_mode(analysis_file, posterior, base_directory='./', optimizations=3, start_point=None, from_mcmc=None, seed=None):
+    '''
+    Finds the mode of the named posterior.
+
+    The optimization process can be initialized either with a random point,
+    a provided parameter point, or by extracting the point with the largest posterior
+    from among previously obtained MCMC samples. The optimization can be iterated to
+    increase the accuracy of the result.
+
+    :param analysis_file: The name of the analysis file that describes the named posterior, or an object of class `eos.AnalysisFile`.
+    :type analysis_file: str or `eos.AnalysisFile`
+    :param posterior: The name of the posterior PDF from which to draw the samples.
+    :type posterior: str
+    :param base_directory: The base directory for the storage of data files. Can also be set via the EOS_BASE_DIRECTORY environment variable.
+    :type base_directory: str, optional
+    :param optimizations: The number of call the the optimization algorithm.
+    :type optimizations: int, optional
+    :param start_point: If provided, the optimization will start at this point.
+    :type start_point: `numpy.ndarray`, optional
+    :param from_mcmc: If provided, the optimization will start at the point with the largest posterior of previously computed MCMC samples.
+        The samples are expected to be stored in the `mcmc-XXXX` subdirectories of the `base_directory`, where 'XXXX' is the provided int.
+    :type from_mcmc: int, optional
+    :param seed: If provided, the optimization will start from a random point and the random number generator will be seeded with this value.
+    :type seed: int, optional
+    '''
+    if optimizations < 1:
+        raise ValueError('The number of optimizations should be larger than zero.')
+    if seed and not start_point is None:
+        raise ValueError('The arguments seed and start_point are mutually exclusive')
+    if seed and not from_mcmc is None:
+        raise ValueError('The arguments seed and fron_mcmc are mutually exclusive')
+    if not from_mcmc is None and not start_point is None:
+        raise ValueError('The arguments from_mcmc and start_point are mutually exclusive')
+
+    if type(analysis_file) is not eos.AnalysisFile:
+        _analysis_file = eos.AnalysisFile(analysis_file)
+    else:
+        _analysis_file = analysis_file
+    analysis = _analysis_file.analysis(posterior)
+    min_chi2 = sys.float_info.max
+    gof = None
+    bfp = None
+
+    eos.info(f'Starting minimization in {optimizations} points')
+
+    if not start_point is None:
+        starting_point = _np.array(start_point)
+        eos.info('Starting optimization from user-provided point')
+
+        _bfp = analysis.optimize(start_point = starting_point)
+        _gof = eos.GoodnessOfFit(analysis._log_posterior)
+        _chi2 = _gof.total_chi_square()
+    elif from_mcmc >= 0:
```

Comment:
If we use a numerical value to id the markov chain, then we should call it the same as in the ``sample_mcmc`` task.
That would be ``chain``.

Reply:
Yes